### PR TITLE
Added token ts docs

### PR DESCRIPTION
**Pull request by jacobcreech**
**Problem**
Many developers going through the spl token documentation are looking for javascript/typescript examples of what is currently shown via the CLI docs. Questions on how to accomplish something the CLI reference shows often come up in the developer support channel.

**Solution**

In this PR, I have provided the javascript/typescript references for all the CLI reference material found on spl.solana.com/token.

You can find the build deployed [here](https://jacobcreech.github.io/solana-program-library/spl/token)

Based on the current open PR https://github.com/solana-labs/solana-program-library/pull/2539

### CODE REVIEW

**Comment by jordansexton** [comment]
Nice! Looks good so far. I'll hold off on full review until the TS PR gets approved. I think there will be some time after it's merged that it needs to be documented before it can replace the npm latest version, but maybe it can be released as `0.2.0-beta.0` or something.

File: docs/src/token.mdx
Comment (joncinque) [inline docs/src/token.mdx:857]: This is actually possible! Strange, but possible. I added a test for it: https://github.com/solana-labs/solana-program-library/pull/2582

File: docs/src/token.mdx
Comment (joncinque) [inline docs/src/token.mdx:1476]: :eyes:

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx:191]: I was trying to match exactly how it would be for the CLI. Is there a specific reason for the default to 9 on the CLI?

Otherwise my comment would be about the default for CLI, which doesn't seem too helpful.

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx]: Updated to `jsx`

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx]: Updated to `*/`

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx]: Updated to `devnet` for consistency

File: docs/src/token.mdx
Comment (joncinque) [reply docs/src/token.mdx]: The CLI likely defaults to 9 because SOL has 9 decimal places.  Either way, some sort of comment will be helpful!

Comment (Jacob-CG) [reply docs/src/token.mdx]: Updated with comment

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx]: Wow, I didn't see that instruction when I went looking for it. Updated with ts example

File: docs/src/token.mdx
Comment (Jacob-CG) [reply docs/src/token.mdx:410]: I don't actually know entirely what happens under the hood. My understanding is as follows:

When you want to wrap SOL, you can send SOL to an associated token account on the native mint and call `syncNative`. `syncNative` updates the `amount` field on the token account to match the amount of wrapped SOL available. That SOL is only retrievable by closing the token account and choosing the desired address to send the token account's lamports.

Is all of that correct? Is there anything I am missing?

Comment (joncinque) [reply docs/src/token.mdx:410]: Yep, you've got it all right.  You just have to create / initialize the token account on the native mint first.

**Comment by jordansexton** [comment]
Currently waiting on https://github.com/solana-labs/solana-program-library/pull/2899. I'll check the docs against the latest API soon.

File: docs/src/token.mdx
Comment (jordansexton) [inline docs/src/token.mdx:106]: `spl-token@latest/lib/index.life.min.js` -- will this link work? I don't know how unpkg works, does it automatically create this file?

**Review by jordansexton — DISMISSED** [review]
Looks good to me!

File: docs/src/token.mdx
Comment (jacobcreech) [reply docs/src/token.mdx:106]: Oh, this no longer works. I'll update to remove it

Comment (jacobcreech) [reply docs/src/token.mdx]: Removed in ed43e1d4010d8f77050477cdb6e5ed2093bc078a

**Pull request merged**
